### DISCUSSION TITLE
Permanent dark window borders?

### DISCUSSION THREAD

**Discussion by johnsonmt01** [SebastJava/mint-themes#51] [Q&A] [answered]
Can gtk.css be modified to always trigger a "dark" window borders theme, even when the base/light theme is selected?  This would be useful in bringing back the older "mixed" style look.

**Comment by SebastJava** [comment]
You are talking about `gtk.css`, so i assume you want to try some direct edit into this file, rather than going through the huge, complex edit and build process as described in https://github.com/SebastJava/mint-yz-theme/blob/master/Do-It-Yourself.md For such a small edit, i understand you might prefer to try a direct edit on `gtk.css`.

Usually, when i want to quickly try things, i do some edits into `gtk.css`, then save, and then force refresh by switching themes back and forth, through GUI or CLI:
```
gsettings set org.cinnamon.desktop.interface gtk-theme 'Mint-Y'
sleep 0.5
gsettings set org.cinnamon.desktop.interface gtk-theme 'Mint-Yz-Base'
```
But this doesn't work for window title bars and backgrounds. You have to logout and login for each tweak you want to try on window borders. I could be wrong about this because i am still on LM Cinnamon 20.3. Well, for most of the time. I also have version 21 fully installed, but i tend to avoid it because there are still some annoyances, some bugs i am still waiting to see fixed...

Okay, that was just a quick introduction, now about your question... I tried a quick cheating. I knew those gray title bars had specific `#HEX` values, so i just color picked it: `#e8e8e8`. Then i did a quick search and replace, and i only replaced the values for `headerbar` and/or `titlebar`. Then i saved the `gtk.css` and i had to logout/login to see the changes take effect.

It worked. But unfortunately i saw i should also change the title text and the window control icons, because those were still dark, on a now dark window border... And i was unlucky on changing these text and icons in a simple way. At that point, i would go through the long, complex https://github.com/SebastJava/mint-yz-theme/blob/master/Do-It-Yourself.md

Additionally, you would have to specify which version of Mint you are using, since things got different from 20.x to 21.x, and from Mate to Cinnamon.

You could find some other help on the forums: https://forums.linuxmint.com/viewforum.php?f=211
You could look at this script and try some mods: https://forums.linuxmint.com/viewtopic.php?f=211&t=382890
If you choose to try my long and complex process, take note that changing the colors is usually the easy part, as they are all grouped into https://github.com/SebastJava/mint-yz-theme/blob/master/src/Mint-Y/gtk-3.0/sass/_colors.scss

**Comment by SebastJava** [comment, marked as answer]
Ah, okay, i found from your other question that you are using this old Cinnamon 4.8.6. So, you have two kinds of window borders. The Metacity window borders and the CSD window borders.

* Metacity window borders: Nemo, Xed, etc. (defined in `Mint-Yz-Base/metacity-1/metacity-theme-3.xml`)
* CSD window borders: Calculator, gnome-disks, etc. (defined in `gtk-3.0/gtk.css`)

As explained before, the CSD window borders seems to be too complex for some "quick hacking". I did succeed in rapidly finding my way to get the backgrounds changed, but then i got stuck when i wanted to change the foreground text and icons, because it was just dark on dark... And this was getting too heavy since i am still on the old LM Cinnamon 20.3, where things are a bit different, for window borders. I had to work in VM, where i don't have all my tools installed, yet. As said before, i guess you would need to got through the big Sass compiling procedure for this: https://github.com/SebastJava/mint-yz-theme/blob/master/Do-It-Yourself.md (Sorry, I am not even trying this at the moment...)

But the window borders for things like Nemo and Xed, on your old Cinnamon version... That's another story. That's Metacity.

1. So, those "generic", "no CSD" window borders are defined in `metacity-1/metacity-theme-3.xml`. But don't touch that one, because the colors are defined in `gtk-3.0/gtk.css`. Search in this `gtk.css` for "@define-color wm_"... "wm" stands for window manager, i guess.
2. For quick and safe tests, you can duplicate your Mint-Yz-Base theme into `~/.themes`. That's a hidden folder. Press Ctrl-H to see it. Create folder if needed.
3. Edit the `gtk-3.0/gtk.css`. You must edit the main Mint-Yz-Base, not some Mint-Yz-Base-\<color\> because they all depend on this "MintClassic" green theme... Search in this `gtk.css` for "@define-color wm_"... Change some `#HEX` values as you see fit. Here is my quick-bad-incomplete test:
```
@define-color wm_title #ffffff;
@define-color wm_title_unfocused #808080;
@define-color wm_bg #202020;
@define-color wm_bg_unfocused #202020;
@define-color wm_highlight #EEEEEE;
@define-color wm_border #BDBDBD;
@define-color wm_border_unfocused #cacaca;
@define-color wm_shadow alpha(black, 0.35);
@define-color wm_button_close_bg #92b372;
@define-color wm_button_close_hover_bg #f20d1f;
@define-color wm_button_close_active_bg #ff791a;
@define-color wm_icon_close_bg #ffffff;
@define-color wm_button_hover_bg #fdfdfd;
@define-color wm_button_active_bg #909090;
@define-color wm_icon_bg #ffffff;
@define-color wm_icon_unfocused_bg #808080;
@define-color wm_icon_hover_bg #404040;
@define-color wm_icon_active_bg #ffffff;
```
4. Save this edited `gtk-3.0/gtk.css`. Switch your windows borders and GTK themes back and forth in your GUI themes selector.
5. If you did your test on a local duplicate in `~/.themes`, and it all works good, you can now force copy your modified `gtk.css` with a DANGEROUS `sudo` edit in the `/usr/share/themes/Mint-Yz-Base/gtk-3.0` directory. And delete your temporary test in `~/.themes`.

**Result:**

![Metacity-hack](https://user-images.githubusercontent.com/31940713/206911104-17d4db79-3388-4c5c-916b-2f8e3675eb33.png)

(That's just a quick and incomplete test...)

**Reply by johnsonmt01** [reply]
Excellent, thank you.  I will experiment with this.  Debian 12 should receive the latest version of Cinnamon, which does not use Metacity.  Perhaps by then we'll discover the secret of fully customizing window manager controls.

**Reply by SebastJava** [reply]
@johnsonmt01 There are new developments there: https://forums.linuxmint.com/viewtopic.php?f=211&t=384994

You could find some interesting informations, if you find the time to scroll through all this topic.
